src/components/SettlementButton.js

```diff
@@ -88,43 +100,64 @@ const defaultProps = {
     },
 };
 
-class SettlementButton extends React.Component {
-    componentDidMount() {
+function SettlementButton({
+    addDebitCardRoute,
+    addBankAccountRoute,
+    anchorAlignment,
+    betas,
+    buttonSize,
+    chatReportID,
+    currency,
+    enablePaymentsRoute,
+    iouReport,
+    isDisabled,
+    isLoading,
+    formattedAmount,
+    nvp_lastPaymentMethod,
+    onPress,
+    policyID,
+    shouldShowPaymentOptions,
+    shouldShowPaypal,
+    style,
+}) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+
+    useEffect(() => {
         PaymentMethods.openWalletPage();
-    }
+    }, []);
 
-    getButtonOptionsFromProps() {
+    const getButtonOptionsFromProps = () => {
         const buttonOptions = [];
-        const isExpenseReport = ReportUtils.isExpenseReport(this.props.iouReport);
+        const isExpenseReport = ReportUtils.isExpenseReport(iouReport);
         const paymentMethods = {
             [CONST.IOU.PAYMENT_TYPE.EXPENSIFY]: {
-                text: this.props.translate('iou.settleExpensify', {formattedAmount: this.props.formattedAmount || ''}),
+                text: translate('iou.settleExpensify', {formattedAmount: formattedAmount || ''}),
```

Comment:
`formattedAmount` is a props, that has a default value `""`. Do we really need have one more default value for this?
```suggestion
                text: translate('iou.settleExpensify', {formattedAmount}),
```